### PR TITLE
docs: Add documentation best-practices / guideline docs

### PR DESCRIPTION
**Pull request by Josmithr**
Note: this PR is not intended to be merged. These documents are intended to be added to the repo wiki. Unfortunately, github does not support PRs for the wiki. Since I'm interested in feedback on the markdown contents, I figured this was a reasonable place to gather it.

The TSDoc guideline docs were adapted from similar documentation I wrote for Whiteboard a few years back. 

Some of the docs (like the notes about abbreviations and such) are an expansion on existing contents on the wiki.

Once folks are happy with the contents, I'll update the wiki.

### CODE REVIEW

File: docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Plans change.
+Even if we intend to add support for some scenario in the future, it may not come to fruition.
+It is best to keep our documentation scoped to only the things we support *right now*.
+
+Note that it's okay to discuss *ideas* for future features or changes, but it needs to be made clear that this is not a promise.
+
+## Don't introduce new terminology
+
+It is imperative that we keep the Fluid Framework's barrier to entry as low as possible to make adoption easier.
```
Comment (skylerjokiel) [inline docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md:18]: Would be great to link to a terminology/glossary page here. We have the external glossary here: https://fluidframework.com/docs/glossary/ but it's focused on external developers so it's not as inclusive as needed for internal developers.

File: docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Please avoid adding new terminology whenever possible.
+If you think you need to add something new to describe some new concept, it will need to be reviewed by the team.
```
Comment (skylerjokiel) [inline docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md:20]: Would be great to provide as much guidance here as possible. Maybe even an alias for now?

File: docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+For a Markdown document, this is a file/page.
+For a source-code comment, this would be a single source-code comment block.
+
+- Remember that our API docs are published for public consumption!
+  So even if this means you have to redefine an acronym multiple times in a code file, it is important to do so for our end-customers.
+
+For example:
+
+```markdown
+The SharedMap is a Distributed Data Structure (DDS) that...
```
Comment (skylerjokiel) [inline docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md:37]: In the casing section below you say: "Distributed data structures", not "Distributed Data Structures"

**Review by skylerjokiel — APPROVED** [review]
Looks good. Can you get @Rick-Kirkham to review also. He's created a lot of the existing ruleset for ff.com docs.

File: docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md (hunk 2 above)
Comment (Josmithr) [reply docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md:23]: @skylerjokiel That sounds like a great idea. Any suggestions for such an alias? Or should we just point folks to the github discussions space? Or both?

**Comment by Josmithr** [comment]
TSDoc guidelines were published: https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines

**Comment by Josmithr** [comment]
JavaScript documentation guidelines were published: https://github.com/microsoft/FluidFramework/wiki/Documenting-JavaScript

**Comment by Josmithr** [comment]
TypeScript documentation guideline landing page was published: https://github.com/microsoft/FluidFramework/wiki/Documenting-TypeScript

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]

File: docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+For a Markdown document, this is a file/page.
+For a source-code comment, this would be a single source-code comment block.
+
+- Remember that our API docs are published for public consumption!
+  So even if this means you have to redefine an acronym multiple times in a code file, it is important to do so for our end-customers.
+
+For example:
+
+```markdown
+The SharedMap is a Distributed data structure (DDS) that...
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply docs-temp-for-PR/Documentation-Dos-And-Do-Nots.md:40]: I had the same thought. This was adapted from existing guidance, but I think it probably merits updating.

File: docs-temp-for-PR/Documenting-JavaScript.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+This includes the set of [tags](https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines/#tsdoc-tags) supported by [TSDoc](https://tsdoc.org/).
+Unless there is a real need to do so, tags should be limited to those supported by TSDoc.
+
+## Exceptions
+
+Notable exceptions to the above statement include:
+
+### Parameter Types
+
+In TypeScript, we don't explicitly document parameter types, since the type system encapsulates that information.
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply docs-temp-for-PR/Documenting-JavaScript.md:13]: It does get automatically added, yes. I can add a note to that effect.

File: docs-temp-for-PR/Fluid-Terminology.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    -   "driver"
+    -   "runtime"
+    -   "host"
+
+-   "Fluid service", not "Fluid server"
+
+## Casing
+
+-   "Container code", not "Container Code"
+-   "Distributed data structures", not "Distributed Data Structures"
```
[the remark this answers is not in the document]
Comment (Josmithr) [reply docs-temp-for-PR/Fluid-Terminology.md:21]: This was adapted from an existing doc. But I agree - this is a proper noun. I'm going to remove this guidance.

**Pull request closed without merging**